### PR TITLE
Fix assertion fallback

### PR DESCRIPTION
**Pull request by Calpicow**
This fixes the case where all xmlns are contained within the top level element.

### CODE REVIEW

**Comment by ericchiang** [comment]
@Calpicow do you have a test case which demonstrates this? I'd like to avoid regressions.

**Comment by Calpicow** [comment]
Yes, I'll add some tests.

**Comment by Calpicow** [comment]
Tests added

**Comment by Calpicow** [comment]
Build is failing, not sure if it's my fault or not

**Comment by ericchiang** [comment]
@Calpicow looking into this now

**Comment by Calpicow** [comment]
Tests are passing

File: connector/saml/saml.go [hunk 1]
```diff
@@ -500,8 +501,8 @@ func verify(validator *dsig.ValidationContext, data []byte) (signed []byte, err
 		verified = true
 		doc.SetRoot(transformedResponse)
 	}
-	assertion := response.SelectElement("Assertion")
-	if assertion == nil {
+	assertion, err := etreeutils.NSSelectOne(response, "urn:oasis:names:tc:SAML:2.0:assertion", "Assertion")
```
Comment (ericchiang) [inline connector/saml/saml.go]: Can you comment a little more on what this change is doing and why it's required?

**Comment by Calpicow** [comment]
Updated. How's that?

**Comment by ericchiang** [comment]
perfect, thanks.

**Pull request merged**